cfme/utils/template/base.py

```diff
@@ -234,19 +241,46 @@ def deploy_template(self):
 
     @log_wrap("download image locally")
     def download_image(self):
+        suffix = re.compile(
+            r'^.*?[.](?P<ext>tar\.gz|tar\.bz2|\w+)$').match(self.image_name).group('ext')
         # Check if file exists already:
         if path.isfile(self.local_file_path):
             logger.info('Local image found, skipping download: %s', self.local_file_path)
-            return True
+            if suffix not in ['zip', 'tar.gz']:
+                return True
+        else:
+            # Download file to cli-tool-client
+            try:
+                request.urlretrieve(self.raw_image_url, self.local_file_path)
+            except URLError:
+                logger.exception('Failed download of image using urllib')
+                return False
 
-        # Download file to cli-tool-client
+        # Unzips image  when suffix is zip or tar.gz and then changes image name to extracted one.
+        # For EC2 and SCVMM images is zip used and for GCE is tar.gz used.
+
+        archive_path = self.image_name
         try:
-            request.urlretrieve(self.raw_image_url, self.local_file_path)
-        except URLError:
-            logger.exception('Failed download of image using urllib')
-            return False
-        else:
+            if suffix == 'zip':
+                archive = ZipFile(archive_path)
+                zipinfo = archive.infolist()
+                self._unzipped_file = zipinfo[0].filename
+            elif suffix == 'tar.gz':
+                archive = tarfile.open(archive_path, "r:gz")
+                self._unzipped_file = archive.firstmember.name
+            else:
+                return True
```

Comment:
I don't necessarily like this pattern of burying the return when suffix isn't of the archive types.

Again I think suffix should be checking against the downloaded file here, not the image_name that's parsed from URL.

May I suggest a bit different structure, that returns more immediately when there's no need to further process the file, instead of doing so within a logic block that is also removing files and re-naming object attributes.

Something along the lines of:

``` python
ARCHIVE_TYPES = ['zip', 'tar.gz']  # should be class constant
if suffix_from_downloaded_file not in ARCHIVE_TYPES:
    return True
else:
    logger.info('Unpacking archived image: %s', self.image_name)
    try:
        ## diapered try block for creating archive
    except Exception:
         ## diaper message
         return False
    try:
         ## big diapered try block for removing the downloaded archive, regardless of the creation failing (re-download next time anyway)
    except Exception:
        ## diaper message
        return False
    try:
         # diapered try block for extracting, closing archive, returning True
     except Exception:
          # diaper message
          return False